src/utils/formHelperFunctions.js

```diff
@@ -293,20 +300,37 @@ const nextStepKey = (nextConditions, metadata, steps, fieldValues) => {
             else {
                 let rulesMet = true;
                 cond.rules.forEach((rule) => {
-                    const ruleVal = rule.value || '';
                     const userVal = fieldValues[rule.key] || '';
-                    let equal;
-                    if (Array.isArray(userVal))
-                        equal = userVal.includes(ruleVal);
-                    else equal = userVal === ruleVal;
-                    rulesMet &=
-                        (equal && rule.comparison === 'equal') ||
-                        (!equal && rule.comparison === 'not_equal');
+                    if (
+                        Array.isArray(userVal) &&
+                        userVal.includes(rule.value)
+                    ) {
+                        newSequence.push(cond.next_step_key);
+                        // array rules are handled after this loop
+                        rulesMet = false;
+                    } else {
+                        const ruleVal = rule.value || '';
+                        const equal = userVal === ruleVal;
+                        rulesMet &=
+                            (equal && rule.comparison === 'equal') ||
+                            (!equal && rule.comparison === 'not_equal');
+                    }
                 });
                 if (rulesMet) newKey = cond.next_step_key;
             }
         });
-    return newKey || defaultKey;
+
+    let nextKey = newKey || defaultKey;
+    // Don't propagate array rules if an equality rule matches or there are no array rules
+    if (newSequence.length === 0 || nextKey) newSequence = stepSequence;
+
+    if ([nextKey].includes(newSequence[0])) {
```

Comment:
Wouldn't this just be `nextKey === newSequence[0]`

Reply:
rip yes lol